packages/bitcoin-dlc-provider/lib/BitcoinDlcProvider.ts

```diff
@@ -371,18 +433,45 @@ export default class BitcoinDlcProvider
     totalCollateral: bigint,
   ) {
     switch (contractDescriptor.type) {
-      case MessageType.ContractDescriptorV0:
+      case MessageType.ContractDescriptorV0: {
         throw Error('ContractDescriptorV0 not yet supported');
+      }
       case MessageType.ContractDescriptorV1:
-        // eslint-disable-next-line no-case-declarations
-        return this.GetPayoutsFromPayoutFunction(
-          dlcOffer,
-          contractDescriptor as ContractDescriptorV1,
-          oracleInfo,
-          totalCollateral,
-        );
-      default:
+        {
+          const contractDescriptorV1 = contractDescriptor as ContractDescriptorV1;
+          const payoutFunction = contractDescriptorV1.payoutFunction as PayoutFunctionV0;
+
+          // TODO: add a better check for this
+          const payoutCurvePiece = payoutFunction.pieces[0].payoutCurvePiece;
+
+          switch (payoutCurvePiece.type) {
+            case MessageType.HyperbolaPayoutCurvePiece:
+              return this.GetPayoutsFromPayoutFunction(
+                dlcOffer,
+                contractDescriptor as ContractDescriptorV1,
+                oracleInfo,
+                totalCollateral,
+              );
+            case MessageType.OldHyperbolaPayoutCurvePiece:
```

Comment:
We should remove in a future PR

(Added issue for it here: https://github.com/AtomicFinance/node-dlc/issues/165)